rotkehlchen/exchanges/coinbaseprime.py

```diff
@@ -0,0 +1,458 @@
+
+        return result  # type: ignore  # mypy doesn't detect that the return type here is defined by the function used
+
+    def _get_portfolio_ids(self) -> list[str]:
+        """
+        Get id of the different portfolios linked to the api keys.
+        May raise: RemoteError
+        """
+        data: dict[str, list] = self._api_query(module='portfolios')
+        try:
+            ids = [portfolio['id'] for portfolio in data.get('portfolios', [])]
+        except KeyError:
+            log.error('Malformed porfolios ids response in coinbase prime. Missing key id')
+
+        return ids
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        try:
+            portfolio_ids = self._get_portfolio_ids()
+        except RemoteError as e:
+            msg_prefix = 'Coinbase Prime API request failed.'
+            msg = (
+                'Coinbase Prime API request failed. Could not reach coinbase due '
+                f'to {e}'
+            )
+            log.error(f'{msg_prefix} Could not reach coinbase due to {e}')
+            return None, f'{msg_prefix} Check logs for more details'
+
+        returned_balances: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        for account_id in portfolio_ids:
+            try:
+                balances_query: dict[str, list[dict[str, Any]]] = self._api_query(
+                    module='portfolios',
+                    path=f'{account_id}/balances',
+                )
+            except RemoteError as e:
+                log.error(f'Failed to query CoinbasePrime balances due to {e}')
+                return None, 'Request to CoinbasePrime failed to fetch balances'
+
+            for balance_entry in balances_query['balances']:
+                try:
+                    total_balance = ZERO
+                    for balance_key in (
+                        'amount',  # includes holds fields as per the docs
+                        'bonded_amount',
+                        'reserved_amount'  # Amount that must remain in the wallet due to the protocol, in whole units  # noqa: E501
+                        'unbonding_amount',
+                        'unvested_amount',
+                        'pending_rewards_amount',
+                    ):
+                        total_balance += deserialize_asset_amount(
+                            amount=balance_entry.get(balance_key, ZERO),
+                        )
+
+                    # ignore empty balances. Coinbase returns zero balances for everything
+                    # a user does not own
+                    if total_balance == ZERO:
+                        continue
+
+                    asset = asset_from_coinbase(balance_entry['symbol'])
+                    try:
+                        usd_price = Inquirer.find_usd_price(asset=asset)
+                    except RemoteError as e:
+                        self.msg_aggregator.add_error(
```

Comment:
Do we really need to add an aggregator error in those? It's just an oracle error which can happen. Shuldn't we just log and continue?